src/integral.jl

```diff
@@ -2,31 +2,41 @@
 # based on https://dlmf.nist.gov/11.13.iii, works for real and complex arguments
 # fails at z=0 because Struve K and Bessel Y diverge there
 
-using SpecialFunctions
-using QuadGK
-
 integrate(f, a, b) = QuadGK.quadgk(f, a, b)[1]
 
 # only valid for real(z) > 0, https://dlmf.nist.gov/11.5.E2
-_K0_integral(z::Real) = (2 / pi) * integrate(t -> exp(-z * sinh(t)), 0, Inf)
-_K_integral(ν, z::Real) = 2(0.5z)^ν / (sqrt(pi) * gamma(ν + 0.5)) *
-    integrate(t -> exp(-z * t) * (1 + t^2)^(ν - 0.5), 0, Inf)
+
+_K0_integral(z::T) where T = (2 / T(pi)) * integrate(t -> exp(-z * sinh(t)), zero(T), T(Inf))
+
+function _K_integral(ν, z::T) where T
+    out = 2 * (z / 2)^ν / (sqrt(T(π)) * gamma(ν + one(T)/2))
+    return out * integrate(t -> exp(-z * t) * (one(T) + t^2)^(ν - one(T)/2), zero(T), T(Inf))
+end
+
 _H0_integral(z::Real) = _K0_integral(z) + bessely0(z)
 _H_integral(ν, z::Real) = _K_integral(ν, z) + bessely(ν, z)
 
 # only valid for real(ν) > -0.5, https://dlmf.nist.gov/11.5.E1
-_H0_integral(z::Complex{T}) where {T <: Real} = (2 / pi) *
-    integrate(ϑ -> sin(z * cos(ϑ)), 0, 0.5pi)
-_H_integral(ν, z::Complex{T}) where {T <: Real} = 2(0.5z)^ν /
-    (sqrt(pi) * gamma(ν + 0.5)) *
-    integrate(t -> (1 - t^2)^(ν - 0.5) * sin(z * t), 0, 1)
+function _H0_integral(z::Complex{T}) where {T <: Real}
+    return (2 / T(π)) * integrate(ϑ -> sin(z * cos(ϑ)), zero(T), T(π)/2)
+end
+
+function _H_integral(ν, z::Complex{T}) where {T <: Real} 
+    out = 2 * (z / 2)^ν / (sqrt(T(π)) * gamma(ν + one(T)/2))
+    return out * integrate(t -> (1 - t^2)^(ν - one(T)/2) * sin(z * t), zero(T), one(T))
+end
+
 _K0_integral(z::Complex{T}) where {T <: Real} = _H0_integral(z) - bessely0(z)
 _K_integral(ν, z::Complex{T}) where {T <: Real} = _H_integral(ν, z) - bessely(ν, z)
 
 # only valid for real(ν) > -0.5, https://dlmf.nist.gov/11.5.E4
-_M0_integral(z) = -(2 / pi) * integrate(ϑ -> exp(-z * cos(ϑ)), 0, 0.5pi)
-_M_integral(ν, z) = -2(0.5z)^ν / (sqrt(pi) * gamma(ν + 0.5)) *
-    integrate(t -> exp(-z * t) * (1 - t^2)^(ν - 0.5), 0, 1)
+_M0_integral(z) = -(2 / T(π)) * integrate(ϑ -> exp(-z * cos(ϑ)), zero(T), T(π)/2)
```

Comment:
I think `T` is missing from the method signature here

Reply:
Should be fixed now